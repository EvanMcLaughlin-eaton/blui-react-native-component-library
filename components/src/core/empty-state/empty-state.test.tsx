import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'react-native-paper';
import { wrapIcon, EmptyState } from '..';
import MatIcon from 'react-native-vector-icons/MaterialIcons';
import MatCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { cleanup } from '@testing-library/react-native';
const Line = wrapIcon({
    IconClass: MatCommunityIcon,
    name: 'chart-line-variant',
});

describe('EmptyState Tests ', () => {
    afterEach(cleanup);
    it('Icon Renders', () => {
        const tree = renderer.create(<EmptyState title={'Test'} IconClass={Line} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Only Title Renders', () => {
        const tree = renderer.create(<EmptyState title={'EmptyState'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Description Renders', () => {
        const tree = renderer.create(<EmptyState title={'Test'} description={'Description'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Actions Renders', () => {
        const tree = renderer
            .create(
                <EmptyState
                    title={'Test'}
                    actions={
                        <Button icon={(): JSX.Element => <MatIcon name={'add-circle-outline'} />}>Add Device</Button>
                    }
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Title, actions, Description Renders', () => {
        const tree = renderer
            .create(
                <EmptyState
                    title={'EmptyState'}
                    description={'Description'}
                    actions={
                        <Button icon={(): JSX.Element => <MatIcon name={'add-circle-outline'} />}>Add Device</Button>
                    }
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
