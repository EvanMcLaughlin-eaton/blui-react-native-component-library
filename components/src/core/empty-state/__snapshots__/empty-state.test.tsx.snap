// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Tests  Actions Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "marginTop": 16,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
  <View
    style={
      Array [
        Object {
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Add Device
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`EmptyState Tests  Description Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "marginTop": 16,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#6200ee",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Description
  </Text>
</View>
`;

exports[`EmptyState Tests  Icon Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 100,
        },
        Object {},
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="icon"
  >
    󰞱
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "marginTop": 16,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`EmptyState Tests  Only Title Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "marginTop": 16,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    EmptyState
  </Text>
</View>
`;

exports[`EmptyState Tests  Title, actions, Description Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "marginTop": 16,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    EmptyState
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#6200ee",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {},
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
        },
        Object {},
        undefined,
        Array [
          Object {
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Description
  </Text>
  <View
    style={
      Array [
        Object {
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Add Device
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
